### DISCUSSION TITLE
How much controller should we use ?

### DISCUSSION THREAD

**Discussion by Elmatbaisaad** [symfony/symfony#46322] [Q&A] [answered]
<h2> Hello everyone </h2>
I m having that question that is it better to create one controller that will manage all the views for example me i have three views or its better to have for each view a controller.

thank you .

**Comment by chapterjason** [comment, marked as answer]
Hey @Elmatbaisaad 

There are multiple approaches to structure your controllers.

## Thematic Controller

This is - I would say - the classic one which will also be generated by the [MakerBundle](https://symfony.com/bundles/SymfonyMakerBundle/current/index.html).
Basically you have all actions for a single area like an entity in a single file.

For example:

```php
<?php

// [...]

class PostController {
    public function index(){}
    public function view(){}
    public function new(){}
    public function delete(){}
}
```

As you can see the controller handles all the actions related to the post entity.

## Thematic Folders with Invokable Controllers

You can also create [invokable controllers](https://symfony.com/doc/current/controller/service.html#invokable-controllers) and create sub folders to keep the responsibility separated.

For example:

```
Controller/
├─ Post/
│  ├─ IndexController.php
│  ├─ NewController.php
│  ├─ ViewController.php
│  ├─ DeleteController.php
```

```php
<?php 

// [...]

class IndexController
{
    public function __invoke(){}
}
```

---

In the end, it's up to you how you want to structure the folders or controllers.
You can also mix both approaches.

**Reply by Elmatbaisaad** [reply]
thank you
